File: connectivity/source/drivers/jdbc/Exception.cxx
/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
/*
 * This file is part of the LibreOffice project.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * This file incorporates work covered by the following license notice:
 *
 *   Licensed to the Apache Software Foundation (ASF) under one or more
 *   contributor license agreements. See the NOTICE file distributed
 *   with this work for additional information regarding copyright
 *   ownership. The ASF licenses this file to you under the Apache
 *   License, Version 2.0 (the "License"); you may not use this file
 *   except in compliance with the License. You may obtain a copy of
 *   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 */

#include "java/lang/Exception.hxx"
using namespace connectivity;

//************ Class: java.lang.Exception


jclass java_lang_Exception::theClass = nullptr;

java_lang_Exception::~java_lang_Exception()
{}

jclass java_lang_Exception::getMyClass() const
{
    // the class must be fetched only once, therefore static
    if( !theClass )
        theClass = findMyClass("java/lang/Exception");
    return theClass;
}

/* vim:set shiftwidth=4 softtabstop=4 expandtab: */

File: net_ure/source/basetypes/Exception.cs
/*
 * This file is part of the LibreOffice project.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

using System;

namespace com.sun.star.uno
{
    public class Exception : System.Exception
    {
        public new string Message { get; set; }
        public IQueryInterface Context { get; set; }
        
        public Exception() { }
        public Exception(string Message, IQueryInterface Context)
        {
            this.Message = Message;
            this.Context = Context;
        }
    }
}
